### PR TITLE
Indexer Subsystem stubs

### PR DESCRIPTION
**Pull request by thehappycamden**
Wrote stubs for indexer API

### CODE REVIEW

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
+package com.spartronics4915.frc2020.subsystems;
+
+import com.spartronics4915.lib.subsystems.SpartronicsSubsystem;
+
+public class Indexer extends SpartronicsSubsystem {
+
+    public int mSpinnerId = -1;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: These should be in Constants.java as `public static final int kSubsystemDescriptorID`.

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    public int rotateClockwiseEmpty() {
+        return 0;
+    }
+
+    public int rotateEmpty() {
+        return 0;
+    }
+
+    public void stop() 
+    {
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Bracket styles

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    public int rotateCounterClockwiseFull() {
+        return 0;
+    }
+
+    public int rotateClockwiseFull() {
+        return 0;
+    }
+
+    public int rotateFull() {}
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Extra closing bracket

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    public void rotateClockwiseOnce() {
+
+    }
+
+    public void rotateCounterClockwiseOnce() {
+
+    }
+
+    public int rotateCounterClockwiseFull() {
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Why do you return `int`s for these?

**Review by omentic — CHANGES_REQUESTED** [review]

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java (hunk 1 above)
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: Remember to add all devices IDs to the [2020 control map](https://docs.google.com/spreadsheets/d/14O3Dj1Bl6tqiHFrCJ7wyVRDOqS5EZ11p4q9j2rYS-3Q/edit#gid=0).

File: src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java (hunk 4 above)
Comment (thehappycamden) [reply src/main/java/com/spartronics4915/frc2020/subsystems/Indexer.java]: number of quarter rotations

File: src/main/java/com/spartronics4915/frc2020/Constants.java [hunk 5]
```diff
@@ -1,4 +1,7 @@
 package com.spartronics4915.frc2020;
 
 public final class Constants {
+    public static final int kIndexerSpinnerId = -1;
```
Comment (omentic) [inline src/main/java/com/spartronics4915/frc2020/Constants.java]: Our new convention is to have all subsystem variables be in their own subclass, so it'll look like this (and `Indexer` is stricken from the name): 

```
public final class Constants {
    public static final class Indexer { 
        public static final int kSpinnerId = -1;
         ... other variables
    }
    // ... other subclasses
}
```

**Review by omentic — CHANGES_REQUESTED** [review]

**Review by omentic — APPROVED** [review]

**Pull request merged**